syntax = "proto3";

package org.openkilda.server42.control.messaging.flowrtt;

import "google/protobuf/any.proto";

message Flow {
    string flow_id = 1;
    enum EncapsulationType {
        VLAN = 0;
        VXLAN = 1;
    }
    EncapsulationType encapsulation_type = 2 [deprecated = true];
    int64 tunnel_id = 3;
    // TODO: rename for avoiding colliding
    EncapsulationType transit_encapsulation_type = 4 [deprecated = true];
    int64 transit_tunnel_id = 5;
    bool direction = 6;
    string dst_mac = 7;
    uint32 udp_src_port = 8;
    int32 hash_code = 9;
    int64 inner_tunnel_id = 10;
    string switch_id = 11;
}

message AddFlow {
    Flow flow = 1;
}

message RemoveFlow {
    Flow flow = 1;
}

message ClearFlowsFilter {
    string dst_mac = 1;
}

message ListFlows {
    repeated Flow flow = 1;
}

message ListFlowsFilter {
    string dst_mac = 1;
}